ow_gazebo_plugins/src/BalovnevModelPlugin/BalovnevModelPlugin.cpp

```diff
@@ -236,26 +239,46 @@ void BalovnevModelPlugin::onModDiffVisualMsg(
     return;
   }
   
-  auto rows = image_handle->image.rows;
-  auto cols = image_handle->image.cols;
+  const auto rows = image_handle->image.rows;
+  const auto cols = image_handle->image.cols;
   if (rows <= 0 || cols <= 0) {
     ROS_WARN("Differential image dimensions are zero or negative");
     return;
   }
 
+  // mask out trench edges to the left and right of the scoop
+  const float EDGE_TRIM_RATIO = 0.3f;
+  const cv::Point2f center(static_cast<float>(rows) / 2.0,
+                           static_cast<float>(cols) / 2.0);
+  const cv::Point2f size(static_cast<float>(rows) * 2.0,
+                         static_cast<float>(cols) * (1.0 - EDGE_TRIM_RATIO));
+  // find angle of scoop relative to image coordinates
+  Vector3 scoop_heading{m_link->WorldPose().Rot().RotateVector(SCOOP_FORWARD)};
+  float yaw = std::atan2(scoop_heading.Y(), scoop_heading.X());
+  const cv::RotatedRect roi(center, size, yaw * RAD2DEG);
```

Comment:
I don't know what roi means. A more descriptive variable name would be nice.

Reply:
Changed its name to `region_of_interest`